samcli/commands/_utils/template.py

```diff
@@ -132,6 +136,22 @@ def _update_relative_paths(template_dict,
 
     """
 
+    for resource_type, properties in template_dict.get("Metadata", {}).items():
+
+        if resource_type not in _METADATA_WITH_LOCAL_PATHS:
+            # Unknown resource. Skipping
+            continue
+
+        for path_prop_name in _METADATA_WITH_LOCAL_PATHS[resource_type]:
+            path = properties.get(path_prop_name)
```

Comment:
Can we abstract this into a function? This looks like the meat of what we do for the Resources.

Reply:
While it might look similar, there are slight differences in the code, as the YAML structure differs below the `Metadata` key and the `Resources` key. The main difference is that the resource type for resources is the value of the `Type` key, while it's a key itself for the resources in the metadata. If you have a good idea how to factor that out into a generic function I'd be happy to do that. :slightly_smiling_face: